examples/causal_language_modeling/mlflow_peft_lora_clm_accelerate_ds_zero3_offload.py

```diff
@@ -249,109 +251,116 @@ def test_preprocess_function(examples):
     if getattr(accelerator.state, "deepspeed_plugin", None):
         is_ds_zero_3 = accelerator.state.deepspeed_plugin.zero_stage == 3
 
-
-    with mlflow_runner:
-        for epoch in range(num_epochs):
-            with TorchTracemalloc() as tracemalloc:
-                model.train()
-                total_loss = 0
-                start_time = time.time()  # Start time for the epoch
-                for step, batch in enumerate(tqdm(train_dataloader)):
-                    outputs = model(**batch)
-                    loss = outputs.loss
-                    total_loss += loss.detach().float()
-                    accelerator.backward(loss)
-                    optimizer.step()
-                    lr_scheduler.step()
-                    optimizer.zero_grad()
-                    
-                end_time = time.time()  # End time for the epoch
-
-                # Calculate metrics
-                epoch_runtime = end_time - start_time
-                samples_per_second = len(train_dataloader) / epoch_runtime
-                steps_per_second = len(train_dataloader) / epoch_runtime
-                avg_loss = total_loss / len(train_dataloader)
-
-                # Log metrics for the epoch
-                mlflow.log_metric('loss', avg_loss)
-                mlflow.log_metric('total_loss', total_loss)
-                mlflow.log_metric('train_runtime', epoch_runtime)
-                mlflow.log_metric('train_samples_per_second', samples_per_second)
-                mlflow.log_metric('train_steps_per_second', steps_per_second)
-            
-            mlflow.end_run()
-
-
-            # Printing the GPU memory usage details such as allocated memory, peak memory, and total memory usage
-            accelerator.print("GPU Memory before entering the train : {}".format(b2mb(tracemalloc.begin)))
-            accelerator.print("GPU Memory consumed at the end of the train (end-begin): {}".format(tracemalloc.used))
-            accelerator.print("GPU Peak Memory consumed during the train (max-begin): {}".format(tracemalloc.peaked))
-            accelerator.print(
-                "GPU Total Peak Memory consumed during the train (max): {}".format(
-                    tracemalloc.peaked + b2mb(tracemalloc.begin)
-                )
+    mlflow.start_run()
+    start_time = time.time()
+    for epoch in range(num_epochs):
+        interval_start_time = time.time()
+        with TorchTracemalloc() as tracemalloc:
```

Comment:
I'm not sure if TorchTracemalloc works well with our framework. Have you checked it as well?

Reply:
As long as we have already implemented GPU Mem parser within ModelRunner, I doubt that it is necessary.

Reply:
The original code have TorchTracemalloc and GPU Mem parser so I kept them